packages/shared/src/components/post/SquadPostPageNavigation.tsx

```diff
@@ -28,7 +34,24 @@ function SquadPostPageNavigation({
   squadLink,
 }: SquadPostPageNavigationProps): ReactElement {
   const isFromNavigation = checkIsFromNavigation();
+  const router = useRouter();
   const link = isFromNavigation ? NOTIFICATIONS_LINK : squadLink;
+  const layout = useFeature(feature.feedLayout);
+
+  if (layout === FeedLayout.V1) {
```

Comment:
For squad it should still always show though right? No matter v1 or not

Reply:
Oh see it below, should we not uniform those?

Reply:
Yeah, was thinking of it too, though the only overlap they had was the history check and I didn't want to include the logic of `routedFromSquad` to be part of the other implementations. Wdyt?

Reply:
Yeah I was indeed thinking about a generic `PostBackButton` which could hold both logic parts?
Let me know if it's a quick win or not.
Seems easier to maintain

Reply:
Makes sense. Let's do it. Would also lessen some logic from the original files.